pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/datasource/BaseDataSource.java

```diff
@@ -127,6 +127,7 @@ public VectorIndexReader getVectorIndex() {
     return getIndex(StandardIndexes.vector());
   }
 
+  @Deprecated
```

Comment:
Given `MapIndexType` is deleted, consider removing all related classes and methods

Reply:
^^